src/helpers/tests/formatAmount.test.ts

```diff
@@ -1,86 +1,108 @@
       '10000000000'
     );
-
     expect(formatAmount({ input: '1000000000000000000', decimals: 4 })).toBe(
       '100000000000000'
     );
-
-    expect(
-      formatAmount({ input: '56817349973594872345', decimals: 18, digits: 4 })
-    ).toBe('56.8173');
   });
+});
 
-  test('handles custom digits', () => {
-    expect(formatAmount({ input: '1000000000000000000', digits: 2 })).toBe('1');
-    expect(formatAmount({ input: '1000000000000000000', digits: 4 })).toBe('1');
+describe('formatAmount – decimal formatting', () => {
+  it('pads or trims to `digits` when showLastNonZeroDecimal is true', () => {
+    expect(
+      formatAmount({
+        input: '1100000000000000000',
+        digits: 4,
+        showLastNonZeroDecimal: true
+      })
+    ).toBe('1.1000');
+    expect(
+      formatAmount({
+        input: '50500000000000000',
+        decimals: 18,
+        digits: 4,
+        showLastNonZeroDecimal: true
+      })
+    ).toBe('0.0505');
   });
 
-  test('adds commas when specified', () => {
+  it('trims trailing zeros when showLastNonZeroDecimal is false', () => {
     expect(
-      formatAmount({ input: '1000000000000000000000', addCommas: true })
-    ).toBe('1,000');
+      formatAmount({
+        input: '1100000000000000000',
+        digits: 4,
+        showLastNonZeroDecimal: false
+      })
+    ).toBe('1.1');
+    expect(
+      formatAmount({
+        input: '1230000000000000000',
+        decimals: 18,
+        digits: 4,
+        showLastNonZeroDecimal: false
+      })
+    ).toBe('1.23');
+  });
 
+  it('limits decimal length to `digits` when too long', () => {
     expect(
       formatAmount({
-        input: '1000000000000000000000',
-        addCommas: true,
-        digits: 2
+        input: '56817349973594872345',
+        decimals: 18,
+        digits: 4,
+        showLastNonZeroDecimal: true
       })
-    ).toBe('1,000');
+    ).toBe('56.8173');
   });
+});
 
-  test('handles showIsLessThanDecimalsLabel', () => {
-    const input = '1000000000000000';
+describe('formatAmount – special behaviors', () => {
+  it('displays less-than label for tiny amounts', () => {
     expect(
       formatAmount({
-        input,
+        input: '1',
+        decimals: 18,
+        digits: 4,
         showIsLessThanDecimalsLabel: true,
-        digits: 2
+        showLastNonZeroDecimal: false
       })
-    ).toBe('<0.01');
+    ).toBe('<0.0001');
   });
 
-  test('handles showLastNonZeroDecimal', () => {
+  it('adds commas to large numbers', () => {
     expect(
       formatAmount({
-        input: '1100000000000000000',
-        showLastNonZeroDecimal: true,
-        digits: 4
+        input: '1000000000000000000000',
+        addCommas: true
       })
-    ).toBe('1.1000');
-
+    ).toBe('1,000');
     expect(
       formatAmount({
-        input: '1100000000000000000',
-        showLastNonZeroDecimal: false,
-        digits: 4
+        input: '1000000000000000000000',
```

Comment:
something is strange about this test.Because decimals is inferred, the test is misleading. One cannot compute mentally what is going on